src/ray/raylet/scheduling_resources.cc

```diff
@@ -9,16 +9,39 @@ namespace ray {
 
 namespace raylet {
 
+bool EqualsZeroEpsilon(double quantity) {
+  if ((quantity <= EPSILON) && (quantity >= -1 * EPSILON)) {
+    return true;
+  }
+  return false;
+}
+
+bool EqualsOneEpsilon(double quantity) {
+  if ((quantity <= 1 + EPSILON) && (quantity >= 1 - EPSILON)) {
+    return true;
+  }
+  return false;
+}
+
 ResourceSet::ResourceSet() {}
 
 ResourceSet::ResourceSet(const std::unordered_map<std::string, double> &resource_map)
-    : resource_capacity_(resource_map) {}
+    : resource_capacity_(resource_map) {
+  for (auto const &resource_pair : resource_capacity_) {
+    RAY_CHECK(resource_pair.second > 0 + EPSILON)
```

Comment:
Comparing 2 double numbers shouldn't add the `EPSILON `.
`RAY_CHECK(resource_pair.second > 0)` is okay.

Let me know if i'm wrong.

Reply:
ignore this.